client/components/Login.jsx

```diff
@@ -1,20 +1,59 @@
 import React from 'react'
+import { Redirect } from 'react-router-dom'
+import { connect } from 'react-redux'
+import { signin } from '../actions/auth'
 
-export default function Login() {
-  return (
-    <div>
-      <h1 className="login">Login</h1>
-      <form className="form">
-        <span>Username </span>
-        <input type="text" name="username" />
-        <br />
-        <br />
-        <span>Password </span>
-        <input type="password" name="password" />
-        <br />
-        <br />
-        <input type="submit" value="Submit" />
-      </form>
-    </div>
-  )
+class Login extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      name: '',
+      surname: '',
+      password: ''
+    }
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  handleChange (e) {
+    const { name, value } = e.target
+    this.setState({
+      [name]: value
+    })
+  }
+
+  handleSubmit (e) {
+    const user = this.state
+    this.props.dispatch(signin(user))
+    this.setState({
+      name: '',
+      surname: '',
+      password: '' })
+    e.preventDefault()
+  }
+
+  render () {
+    if (this.props.auth.loggedIn) {
+      return <Redirect to='/' />
+    }
```

Comment:
I wonder if this conditional would be better placed inside of `componentDidMount`?